### PR TITLE
renderer_vulkan: Import host memory for screenshots

### PR DESCRIPTION
**Pull request by Wunkolo**
When taking a screenshot, an intermediate staging-buffer must be
allocated to copy the frame-data into, and then must be copied yet again
into `settings.screenshot_bits`.

With `VK_EXT_external_memory_host`, `settings.screenshot_bits` can be
directly imported into a `vk::DeviceMemory` and copied into directly by
the driver, allowing for some memory and latency savings.

`VK_EXT_external_memory_host` can be used for other similar use-cases,
but the first client of this feature is just screenshots for now in particular.

For a series of x6-native-resolution 2400x1440 screenshots(13.18MiB) on my RTX 3090(at PCIE 4.0) I see a uniform decrease in the time it takes to transfer this data by about 1ms. This would also mean a 13.18MiB memory-savings now that a staging-buffer is no longer needed.

VK_EXT_external_memory_host

The speed/memory gains would generally scale with the amount of memory transferred. So I'd imagine x10 resolution would mean even more savings in time/memory.

- - -
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/7132)

### CODE REVIEW

File: src/common/memory_detect.cpp [hunk 1]
```diff
@@ -10,8 +10,10 @@
 #include <sys/types.h>
 #if defined(__APPLE__) || defined(__FreeBSD__)
 #include <sys/sysctl.h>
+#include <unistd.h>
 #elif defined(__linux__)
 #include <sys/sysinfo.h>
+#include <unistd.h>
```
Comment (Steveice10) [inline src/common/memory_detect.cpp:16]: You can just move the `unistd.h` out of the this ifdef chain similar to `sys/types.h`, it's included in all 3 paths.

**Comment by GPUCode** [comment]
From a first view, I think it would be good to separate the two implementations in different functions, for example RenderScreenshotWithHostMemory and RenderScreenshotWithStagingCopy.

File: src/video_core/renderer_vulkan/renderer_vulkan.cpp [hunk 2]
```diff
@@ -969,18 +982,164 @@ void RendererVulkan::RenderScreenshot() {
...
+        device.allocateMemoryUnique(allocation_chain.get());
+
+    vk::StructureChain<vk::BufferCreateInfo, vk::ExternalMemoryBufferCreateInfo> BufferInfo;
+    auto& buffer_info = BufferInfo.get<vk::BufferCreateInfo>();
+    buffer_info.size = aligned_size;
+    buffer_info.usage = vk::BufferUsageFlagBits::eTransferDst;
+    buffer_info.sharingMode = vk::SharingMode::eExclusive;
+
+    auto& buffer_external_info = BufferInfo.get<vk::ExternalMemoryBufferCreateInfo>();
+    buffer_external_info.handleTypes = vk::ExternalMemoryHandleTypeFlagBits::eHostAllocationEXT;
```
Comment (GPUCode) [inline src/video_core/renderer_vulkan/renderer_vulkan.cpp:1075]:
```suggestion
    const vk::StructureChain buffer_info = {
        vk::BufferCreateInfo{
            .size = aligned_size,
            .usage = vk::BufferUsageFlagBits::eTransferDst,
            .sharingMode = vk::SharingMode::eExclusive,
        },
        vk::ExternalMemoryBufferCreateInfo{
            .handleTypes = vk::ExternalMemoryHandleTypeFlagBits::eHostAllocationEXT,
        }
    };
```
ditto for structure chain above

**Review by GPUCode — APPROVED** [review]

**Comment by Steveice10** [comment]
This had some conflicts with the barycentric PR that is now merged, can you rebase so we can canary this?

**Comment by Wunkolo** [comment]
Note to self to revisit this later once [VK_EXT_host_image_copy](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_EXT_host_image_copy.html) comes out.

**Pull request merged**